smart-client/src/main/java/org/smartdata/client/SmartClient.java

```diff
@@ -145,6 +159,38 @@ private void checkOpen() throws IOException {
     }
   }
 
+  /**
+   * Record active server currently found into a local file.
+   */
+  private void recordActiveServerAddress(String address)
+      throws IOException {
+    if (!new File(ACTIVE_SMART_SERVER_FILE_PATH).exists()) {
+      new File(ACTIVE_SMART_SERVER_FILE_PATH).createNewFile();
+    }
+    FileWriter fw = null;
+    try {
+      fw = new FileWriter(ACTIVE_SMART_SERVER_FILE_PATH);
+      fw.write(address);
+    } finally {
+      if (fw != null) {
+        fw.close();
+      }
+    }
+  }
+
+  /**
+   * Get recorded active server address.
+   */
+  private InetSocketAddress getActiveServerAddress() {
+    Scanner scanner = new Scanner(ACTIVE_SMART_SERVER_FILE_PATH);
+    if (scanner.hasNextLine()) {
+      String address = scanner.nextLine();
+      String[] strings = address.split(":");
```

Comment:
This address is like hostname/ip:port, and strings[0] is hostname/ip. It's not correct hostname for InetSocketAddress.